src/main/ngapp/accio/src/app/mask.directive.ts

```diff
@@ -0,0 +1,140 @@
+import { Directive, HostListener, ElementRef, Input } from '@angular/core';
+import { MagicWandService } from './magic-wand.service';
+
+@Directive({
+  selector: '[appMask]',
+})
+export class MaskDirective {
+  @Input() scale: number;
+  @Input() originalData: ImageData;
+
+  // create direct reference of canvas on editor.html
+  constructor(
+    private canvas: ElementRef<HTMLCanvasElement>, 
+    private magicWandService: MagicWandService,
+  ) { }
+  private ctx: CanvasRenderingContext2D;
+  private maskImage: HTMLImageElement;
+
+  @HostListener('click', ['$event'])
+  onClick(e: MouseEvent) {
+    const xCoord = e.offsetX;
+    const yCoord = e.offsetY;
+    this.ctx = this.canvas.nativeElement.getContext('2d');
+    let imgData = this.ctx.getImageData(0,0, this.canvas.nativeElement.width, this.canvas.nativeElement.height);
+
+    // tolerance will come from user input
+    const tolerance = 20;
+
+    // returns an array indicies of each pixel in the mask
+    const maskPixels = this.magicWandService.floodfill(this.originalData, Math.floor(xCoord / this.scale), Math.floor(yCoord / this.scale), tolerance);
+
+    // alpha value hardcoded for now
+    this.drawMask(maskPixels, imgData, 1);
+  }
+
+  /**
+   *  Creates a UI mask to the user by reading in all values in the given maskPixels
+   *  Clears the canvas fisrt so the mask can be saved as a new image and then over-
+   *    lays the mask on top of the scaled image. 
+   *  Adds all pixels to mask by giving alpha value of 255, changing global alpha on 
+   *    maskImage will multiply 255 by globalAlpha = [0,1]
+   *  maskPixels: set of coordinates indexed by red value of pixels to be in mask
+   *  imgData: Uint8clampedArray of all pixel data in scaled image. 
+   *  alphaValue: the number the user passes in on a slide bar for transparency of the mask
+   */
+  private drawMask(maskPixels: Set<number>, imgData: ImageData, alphaValue: number): void {
+    //clear old context
+    this.ctx.clearRect(0,0,this.canvas.nativeElement.width, this.canvas.nativeElement.height);
+    
+    //new imagedata creates transparent black rectangle
+    let img = new ImageData(imgData.width, imgData.height);
+
+    // access all pixels in original mask, scale them up to image on UI width
+    // ***** Need flood fill to pass back total mask, not just new one created, used for undo/redo as well
```

Comment:
nit: clarify comment from ***** notation: ie Is this a NOTE or a TODO?

Reply:
This has to do with your prev comment, I will label them all as TODO-Mask